analyzers/src/SonarAnalyzer.CSharp/Rules/UnconditionalJumpStatement.cs

```diff
@@ -75,15 +70,43 @@ private class LoopWalker : LoopWalkerBase<StatementSyntax, SyntaxKind>
                 SyntaxKind.CatchClause
             };
 
-            public LoopWalker(SyntaxNodeAnalysisContext context, ISet<SyntaxKind> loopStatements)
-                : base(context, loopStatements)
+            public LoopWalker(SyntaxNodeAnalysisContext context, ISet<SyntaxKind> loopStatements) : base(context, loopStatements) { }
+            public override void Visit()
             {
+                var csWalker = new CsLoopwalker(this);
+                csWalker.SafeVisit(rootExpression);
             }
 
-            public override void Visit()
+            protected override bool IsAccessToClassMember(StatementSyntax node)
             {
-                var csWalker = new CsLoopwalker(this);
-                csWalker.SafeVisit(this.rootExpression);
+                var returnStatementExpression = ((ReturnStatementSyntax)node).Expression;
+                if (returnStatementExpression is IdentifierNameSyntax identifier
+                    && semanticModel.GetSymbolInfo(identifier) is { } symbolInfo
+                    && symbolInfo.Symbol is { } symbol
```

Comment:
`SymbolInfo` is a struct so it cannot be `null`. You can go directly for it's symbol
```suggestion
                    && semanticModel.GetSymbolInfo(identifier).Symbol is { } symbol
```